build.gradle.kts

```diff
@@ -51,28 +45,49 @@ repositories {
     maven(url = "https://kotlin.bintray.com/kotlinx")
 }
 
-dependencies {
-    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
-
-    implementation(kotlin("stdlib-jdk8"))
-    implementation(group = "org.snakeyaml", name = "snakeyaml-engine", version = "2.2.1")
-    implementation(group = "org.jetbrains.kotlinx", name = "kotlinx-serialization-core", version = "1.0.1")
-
-    val spekVersion = "2.0.15"
-
-    testImplementation(group = "org.spekframework.spek2", name = "spek-dsl-jvm", version = spekVersion)
-    testImplementation(group = "ch.tutteli.atrium", name = "atrium-fluent-en_GB", version = "0.15.0")
-
-    testRuntimeOnly(group = "org.spekframework.spek2", name = "spek-runner-junit5", version = spekVersion)
-}
-
-tasks.withType<KotlinCompile> {
-    kotlinOptions.jvmTarget = "1.8"
-    kotlinOptions.freeCompilerArgs = listOf("-Xuse-experimental=kotlin.RequiresOptIn")
-}
-
 kotlin {
     explicitApi()
+
+    jvm {
+        withJava()
+        compilations.all {
+            kotlinOptions {
+                jvmTarget = "1.8"
+            }
+        }
+    }
+
+    sourceSets {
+        all {
+            languageSettings.useExperimentalAnnotation("kotlin.RequiresOptIn")
+        }
+        val commonMain by getting {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:${Versions.serialization}")
+            }
+        }
+        val commonTest by getting {
+            dependencies {
+                implementation("org.spekframework.spek2:spek-dsl-metadata:${Versions.spek}")
+                implementation("ch.tutteli.atrium:atrium-fluent-en_GB-common:${Versions.atrium}")
+            }
+        }
+        val jvmMain by getting {
+            dependencies {
+                implementation(project.dependencies.platform("org.jetbrains.kotlin:kotlin-bom"))
+                implementation(kotlin("stdlib-jdk8"))
+                implementation("org.snakeyaml:snakeyaml-engine:${Versions.snakeYaml}")
+                implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:${Versions.serialization}")
+            }
+        }
+        val jvmTest by getting {
+            dependencies {
+                implementation("org.spekframework.spek2:spek-dsl-jvm:${Versions.spek}")
+                implementation("ch.tutteli.atrium:atrium-fluent-en_GB:${Versions.atrium}")
```

Comment:
This is quite the opposite. `jvmTest` does not depend on `commonTest` and it would not make much sense as the JVM implementations of the frameworks are usually more powerful and you don't need conflicting definitions in your classpath. I'd leave it here as it does no harm (version is centralized elsewhere…).